include/flatbuffers/code_generators.h

```diff
@@ -29,7 +29,7 @@ class BaseGenerator {
     EnsureDirExists(path.c_str());
     if (parser.opts.one_file) return path;
     std::string namespace_dir = path;  // Either empty or ends in separator.
-    auto &namespaces = ns.components;
+    auto namespaces = PrefixedNamespaceComponents(ns, parser.opts.lang);
```

Comment:
Changes current address package statement generation, and creating the necessary sub-directories for the namespace prefix.
Thinking about it, namespace prefix be accounted for in import statement generation. I'll tweak some namespace examples and check if other changes are needed...